pkg/reconciler/taskrun/validate_resources.go

```diff
@@ -73,13 +73,29 @@ func validateParams(ctx context.Context, paramSpecs []v1beta1.ParamSpec, params
 	var neededParams []string
 	paramTypes := make(map[string]v1beta1.ParamType)
 	neededParams = make([]string, 0, len(paramSpecs))
+	neededObjectParamsKeys := make(map[string][]string)
 	for _, inputResourceParam := range paramSpecs {
 		neededParams = append(neededParams, inputResourceParam.Name)
 		paramTypes[inputResourceParam.Name] = inputResourceParam.Type
+
+		// collect needed keys for each parameter of type Object
+		if inputResourceParam.Type == v1beta1.ParamTypeObject {
```

Comment:
I'm not the biggest fan that we're interleaving general param validation and object-specific validation here - it makes it harder to follow the code because you need to context switch back between what is / isn't object specific. Could we hoist this the object-specific code into a single func that gets called after we've done the param validation that applies to all params?

Reply:
+1 to that

Reply:
yeah, this will be refactored out to a separate function also discussed [here](https://github.com/tektoncd/pipeline/pull/4786#discussion_r863230699)